HopsanGUI/HcomHandler.cpp

```diff
@@ -6320,107 +6355,139 @@ void HcomHandler::evaluateExpression(QString expr, VariableType desiredType)
     //else if(desiredType != Scalar && expr.startsWith("fft(") && expr.endsWith(")"))
     else if(desiredType != Scalar && isHcomFunctionCall("fft", expr))
     {
-        QString args = expr.mid(4, expr.size()-5);
-        QStringList splitArgs = SymHop::Expression::splitWithRespectToParentheses(args, ',');
-        if(splitArgs.size() == 1)
-        {
-            mAnsType = DataVector;
-            const QString varName = args.section(",",0,0).trimmed();
-            evaluateExpression(varName, DataVector);
-            SharedVectorVariableT pVar = mAnsVector;
-            if (mAnsType == DataVector)
-            {
-                mAnsType = DataVector;
-                mAnsVector = pVar->toFrequencySpectrum(pVar->getSharedTimeOrFrequencyVector(), false);
-                return;
+        QStringList args = expr.mid(4, expr.size()-5).split(",");
+        for(int a=0; a<args.size(); ++a) {
```

Comment:
Why not use "split args with respect to parentheses". What if the arguments are function taking other arguments,
I belive that is what that functions solved, or maybe that was an other one

Reply:
Good point. It is required for using input vector from another vector function, e.g. `fft(lp1(vector))`. I also added proper evaluation of min and max time using the `getNumber()` function.

Reply:
I had forgotten about getNumber, It is likely needed in other places as well, (to do later)